server/src/main/java/org/opensearch/index/engine/NoOpEngine.java

```diff
@@ -209,11 +209,12 @@ public void trimUnreferencedTranslogFiles() throws TranslogException {
                                 translog.trimUnreferencedReaders();
                                 // refresh the translog stats
                                 translogStats = translog.stats();
-                                assert translog.currentFileGeneration() == translog.getMinFileGeneration() : "translog was not trimmed "
-                                    + " current gen "
-                                    + translog.currentFileGeneration()
-                                    + " != min gen "
-                                    + translog.getMinFileGeneration();
+                                assert engineConfig.getIndexSettings().isRemoteTranslogStoreEnabled()
```

Comment:
can we add a comment here as well why it doesn't hold true for remote translog enabled index